src/OrchardCore.Modules/OrchardCore.Html/Handlers/HtmlBodyPartHandler.cs

```diff
@@ -25,12 +26,16 @@ public override Task GetContentItemAspectAsync(ContentItemAspectContext context,
         {
             return context.ForAsync<BodyAspect>(async bodyAspect =>
             {
-                if (_bodyAspect != null)
+                if (bodyAspect != null && part.ContentItem.Id == _contentItemId)
                 {
                     bodyAspect.Body = _bodyAspect;
+
                     return;
                 }
 
+                _bodyAspect = bodyAspect.Body;
+                _contentItemId = part.ContentItem.Id;
```

Comment:
You are setting this cache value here, before it has even been generated. 
So cannot work

Please refer to line 51 and 55 where the bodyAspect is actually being set.

add `_contentItemId = part.ContentItem.Id;` to after the catch block, after line 56